feat(matchmake-extension): Implement NotificationData methods

DaniElectra opened the pull request:
<!--

* Before making a pull request, ensure the changes are for an approved issue.
* If your changes are not for an approved issue, your pull request can and will be rejected.
*
* CHECK https://github.com/PretendoNetwork/REPO_NAME/issues?q=is%3Aopen+is%3Aissue+label%3Aapproved
* FOR APPROVED ISSUES!

-->

Resolves #XXX

### Changes:

<!--

* Describe your changes in as much detail as possible. Make sure to list your changes, as well as the rationale behind them.
* If applicable, include code snippets, images, videos, etc.
*
* If your changes require any database migrations, describe them in detail and leave any migration queries inside of a code
* block within a <details> tag.

-->

**NOTE: The changes currently haven't been tested, so marking as draft**

The NotificationData methods are used by games to send notifications to friends about user activity, among others. These notifications are created or updated using `UpdateNotificationData`, which the server will register and send to the connected friends (as seen on Mario Tennis Open).

The lifetime of these notifications is the same as the connection of the user who sends them. That is, when a user sends a notification and then disconnects, the notifications will be discarded.

All notifications sent over `UpdateNotificationData` are logged inside the `tracking.notification_data` table to prevent abuse. The type of these notifications is also constrained to a range of specific values reserved for game-specific purposes (from 101 to 108).

- [x] I have read and agreed to the [Code of Conduct](https://github.com/PretendoNetwork/Pretendo/blob/master/.github/CODE_OF_CONDUCT.md).
- [x] I have read and complied with the [contributing guidelines](https://github.com/PretendoNetwork/Pretendo/blob/master/.github/CONTRIBUTING.md).
- [ ] What I'm implementing was an [approved issue](../issues?q=is%3Aopen+is%3Aissue+label%3Aapproved).
- [ ] I have tested all of my changes.

DaniElectra committed:
feat(matchmake-extension): Implement NotificationData methods

The NotificationData methods are used by games to send notifications to
friends about user activity, among others. These notifications are
created or updated using `UpdateNotificationData`, which the server will
register and send to the connected friends (as seen on Mario Tennis Open).

The lifetime of these notifications is the same as the connection of the
user who sends them. That is, when a user sends a notification and then
disconnects, the notifications will be discarded.

All notifications sent over `UpdateNotificationData` are logged inside
the `tracking.notification_data` table to prevent abuse. The type of
these notifications is also constrained to a range of specific values
reserved for game-specific purposes (from 101 to 108).

DaniElectra committed:
fix(matchmake-extension): Fix incorrect SQL query for notifications

Also improve and fix the NotificationData methods.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/PretendoNetwork/nex-protocols-common-go?pullRequest=49) <br/>All committers have signed the CLA.

DaniElectra committed:
fix(matchmake-extension): Send notification data to connected friends

Removes warnings for connections not being found when it isn't relevant.

DaniElectra committed:
fix(matchmake-extension): Fix OnAfter typo on notification data methods

DaniElectra committed:
fix(matchmake-extension): Fix notification string length check

Kid Icarus: Uprising sends strings with byte length longer than 156, so
assume this should count runes instead.

DaniElectra commented:
This has been tested on various games and working without issues after fixing some bugs, so merging
